src/mp4box/trun.rs

```diff
@@ -97,10 +108,30 @@ impl<R: Read + Seek> ReadBox<&mut R> for TrunBox {
             None
         };
 
-        let mut sample_durations = Vec::with_capacity(sample_count as usize);
-        let mut sample_sizes = Vec::with_capacity(sample_count as usize);
-        let mut sample_flags = Vec::with_capacity(sample_count as usize);
-        let mut sample_cts = Vec::with_capacity(sample_count as usize);
+        let mut sample_durations = Vec::new();
+        let mut sample_sizes = Vec::new();
+        let mut sample_flags = Vec::new();
+        let mut sample_cts = Vec::new();
+        if u64::from(sample_count) * u64::from(sample_size as u64)
```

Comment:
There's a no-op conversion here: `u64::from(sample_size as u64)`.

Reply:
fixed in [619a36b](https://github.com/privacyresearchgroup/mp4-rust/pull/2/commits/619a36b38b435b09ba997434d810f0eab2ad3b35)